lib/dialogue-agent/conversation.ts

```diff
@@ -548,4 +570,48 @@ export default class Conversation extends events.EventEmitter {
             console.log('Almond sends link: '+ url);
         return this._addMessage({ type: MessageType.LINK, url, title });
     }
+
+    appendLog(turn : DialogueTurn) {
+        this._log.push(turn);
+    }
+
+    upvoteLast() {
```

Comment:
One method, `voteLast` with a vote parameter? Also, I'm not a fan of using emojis for this.